### PR TITLE
make the wifi chest pipeworkable

### PR DESCRIPTION
**Pull request by Desour**
Required https://github.com/minetest-mods/pipeworks/pull/194 to work.

### CODE REVIEW

**Comment by DS-Minetest** [comment]
The pipeworks PR is merged.

**Comment by DS-Minetest** [comment]
Please feedback.

**Comment by thomasrudin** [comment]
@SmallJoker (sorry for pinging) this looks nice, did this fall under the table? :smile:

File: wifi.lua [hunk 1]
```diff
@@ -3,7 +3,46 @@ minetest.register_node("more_chests:wifi", {
...
+				return false
+			end
+			local player = minetest.get_player_by_name(owner)
+			if not player then
+				return false
+			end
+			local inv = player:get_inventory()
+			return inv:room_for_item("more_chests:wifi", stack)
+		end,
+		input_inventory = "more_chests:wifi",
```
Comment (thomasrudin) [inline wifi.lua]: Does that work with detached inventories? :astonished:

**Comment by SmallJoker** [comment]
@thomasrudin Yes, together with many other abandoned PRs it's on the last page of all open minetest-mods PRs. I'd have to look into this mod before merging - even if it's confirmed to be working.

File: wifi.lua (hunk 1 above)
Comment (DS-Minetest) [reply wifi.lua]: If I remember correctly, this here is the listname.
The inventory is gotten via `return_input_invref` (here's the pipeworks commit that made this possible: https://gitlab.com/VanessaE/pipeworks/-/commit/a5e1bfa0e824ab70877803d087c97440ff729cac#38deb38da9132073cae6df77fe1bf77a55727e02_322_323).
Also note that the wifi chest uses player inventories, not detached.

**Comment by Megaf** [comment]
Hi, is this PR still relevant after #21 ?

**Comment by Desour** [comment]
Oh, activity.

Yes, this is still relevant. I just took a look into #21, and I'm sure it doesn't do what this PR does (it doesn't even seem to handle the wifi chest correctly).

For explanation of this PR here:
Pipeworks' devices (eg. filter) give items an owner, ie. the owner of the filter is the owner of the item. This is used to decide in what inventory the item should go to when it arrives at a wifi chest.

**Comment by Desour** [comment]
Squashed and rebased.

**Comment by Megaf** [comment]
@Desour Would it be too hard to make pipework dependency optional?
I don't see pipework as a requirement for this, basically, if pipework is enabled then it connects to it, if it isnt, then it's just there.

Also, is the wifi chest handling this alright?

**Comment by Desour** [comment]
>pipework dependency optional

Oops. Fixed. (And tested.)

>Also, is the wifi chest handling this alright?

Sorry, I do not understand the question. What are you referring to with "this"?

**Comment by Megaf** [comment]
Thanks @Desour!
Don't worry about my question, I misread something.
PR Looks good! Merging!

**Pull request merged**